### DISCUSSION TITLE
T3E v2 (pcb rev. v2.3)

### DISCUSSION THREAD

**Discussion by htvekov** [HASwitchPlate/openHASP#458] [Hardware]
# NEW T3E SKU AVAILABLE !! 🎉

![20230401_232423](https://user-images.githubusercontent.com/45469813/229314529-87852152-d49d-4a62-b227-649f6a5983f8.jpg)

New revised v2 of the [T3E](https://openhasp.haswitchplate.com/0.7.0/hardware/golden-security/gs-t3e/) device is now available. Old v1 has been discontinued.
The device is in exterior completely identical to the [T3E v1](https://github.com/HASwitchPlate/openHASP/discussions/383), [WT86-32-3ZW1](https://github.com/HASwitchPlate/openHASP/discussions/311) and the quite new [WT32S3-86V](https://github.com/HASwitchPlate/openHASP/discussions/435) as the outer casing and dimensions are 100% identical. These devices only differs in the backplate versions/measurements as well as the availability of PSU / relays.

The issue with poor WiFi signal strength on the T3E v1 has been fully resolved 🎉
The previous solution with 'direct mcu solder on pcb' has been substituted with an ESP module with proper internal WiFi antenna.

I would argue this new T3E v2 is the clear winner when it comes to available 4" square display devices 🏆
Easy to flash using USB, good WiFi signal strength, high quality and high resolution IPS display with almost infinite viewing angles, EU sized backplate including PSU and three relays. A perfect 'plug & play' openHASP 4" display device ! 😃

[Alibaba link](https://www.alibaba.com/product-detail/2022-new-arrival-smart-wifi-switch_1600573806214.html?spm=a2700.details.0.0.252d69cf8GeWdp) for the T3E v2 device from known vendor - Golden Security
**Note: Specify that you want the new device with pcb rev. 2.3 !!**

Device is marketed as an in wall multi-function scene panel and a 3 (or 4) gang relay switch. The [T3E](https://www.dropbox.com/s/9li72bxs8s0zvjk/T6E%20VS%20T8E%20and%20T3E.pdf?dl=0) is one of three devices in a series. Both the [T6E & T8E](https://www.dropbox.com/s/9li72bxs8s0zvjk/T6E%20VS%20T8E%20and%20T3E.pdf?dl=0) devices have Tuya zigbee gateway capabilities, where the T3E can only act as a 'slave' device and smart scene panel display. These devices can only be used with the 'Smart Life' / 'Tuya Smart' apps. No local control - only cloud based control through these apps.

**EU backplate version:**
- 3 gang relay switch
- EU backplate 1 gang relay switch + 1 way light dimming version should also be available (0-10V Dimming Wiring Port)
- Max load: 200W pr. gang (600W in total EU version)
- Dimensions: 86x86x36mm
- Device pcb version: v2.3 (20221205)

**PROS:**
- 4" Capacitive HD IPS LCD
  * Resolution: 480 x 480
  * Display driver: [ST7701S](https://ronboelectronics.en.made-in-china.com/product/kdumLsbPArWB/China-Smart-Home-86-Box-4-0-Inch-Resolution-480-X-480-IPS-Capacitive-Touch-Square-LCD-Display.html)
  * Touch controller: Goodix [GT911](https://www.distec.de/fileadmin/pdf/produkte/Touchcontroller/DDGroup/GT911_Datasheet.pdf)
- MCU: [ESP32-S3-WROOM-1](https://www.espressif.com/sites/default/files/documentation/esp32-s3-wroom-1_wroom-1u_datasheet_en.pdf) (Xtensa-LX7 dual core 240MHz)
  * 8 MB PSram - Octal SPI
  * 16 MB flash - Quad SPI
- EU backplate
  * PSU included
  * 3 relays
- Micro USB power connector
- Flashable through USB

**CONS:**
- Tuya device with 'standard' ZS3L Tuya chip on pcb
- ESP32 flash header pins **_not_** populated on pcb  

**`U13` EU backplate connector (marked with `yellow` in picture below):**
Numbered from right (marked with a white dot) to left
1. GPIO45 = L1 relay
2. GPIO46 = L2 relay
3. GPIO44 (RXD0) = L3 relay and `RX`
4. GPIO43 (TXD0) = L4 relay and `TX`
   (**Not** connected in the EU 3 gang relay switch version. L4 & L3 are low voltage outputs in the dimmer version)
6. GND
7. GND
8. VCC (+5V)
9. VCC (+5V)

**`J1` ESP32 flash header (marked with `red` in picture below):**
'Standard' flash header available on the pcb for easy flashing. The actual header pins are **not** populated, so you have to solder these on yourself.
NB ! The white pcb cover plate has holes that match the position of the _previous_ T3E v1 header pins, but only [short profile female header pins](https://www.aliexpress.com/item/10000000838267.html?spm=a2g0o.productlist.main.23.50907bf2A6jEcF&algo_pvid=99e9740c-944e-415a-8edc-e0a91448567a&algo_exp_id=99e9740c-944e-415a-8edc-e0a91448567a-11&pdp_ext_f=%7B%22sku_id%22%3A%2220000000003548910%22%7D) will fit under the cover plate. Standard Dupont male pins won't fit into these female headers. But the commonly used breadboard pins do fit. If you solder in 'regular' male header pins, the backplate can't be fitted, unless you modify the pins in height.

Device can be flashed by either using the USB port or the flash header pins and a FTDI serial programmer device.
Software: E.g. Flash Download Tool v3.9.3
Note that RX & TX are NOT cross wired !
`GND` <--> `GND`
`5V` <--> `5V`
`RX` <--> `RX`
`TX` <--> `TX`
`IO0` pulled low (to GND) during power on to enter boot mode.

**`KEY` button (marked with `orange` in picture below):**
Connected to `EN` on the ESP32-S3 and acts as a `RESET` button

**`U2` connector/pads (marked with `indigo` in picture below):**
**Note: GPIO1 & GPIO2 are both connected to the Tuya chip (U3) !!
So cut 3,3v power trace to the chip or desolder chip completely.
Then these two GPIO's are free to be used for e.g. I2C sensors**

From top to bottom in picture (pad names are covered by the Tuya chip)
VCC: +3.3V
GND: GND
RXD: GPIO2
TXD: GPIO1
NRST: NC (no connection) 

**`P1` USB Micro-B connector (marked with `rosa` in picture below):**
1. VCC: +5V
2. D-: GPIO19
3. D+: GPIO20
4. ID: NC (no connection)
5. GND: GND

**Other useful pin connections identified:**
`GPIO42` = Display backlight (`LCD_BL`)

![T3E_pcb_v23_with_colour_indications](https://user-images.githubusercontent.com/45469813/229310872-dcccd2c0-600e-4317-8542-c04cee9d969d.jpg)

![IMG20230401184414](https://user-images.githubusercontent.com/45469813/229304169-08f67767-cf60-4ef8-93fa-10ef8a009090.jpg)

![IMG20230401184523](https://user-images.githubusercontent.com/45469813/229304183-8c1e55a5-6eef-435e-8690-5ad837d74a70.jpg)

![IMG20230401184428](https://user-images.githubusercontent.com/45469813/229304193-9ee15ab8-021a-4394-9422-35d5af84ed27.jpg)

**Comment by nagyrobi** [comment]
What was the shipping cost?

**Reply by htvekov** [reply]
Too high - 55 usd for 4 devices with UPS.
Cheapest shipping for two devices to EU is appx. 30 usd

**Reply by nagyrobi** [reply]
Thanks

**Comment by fvanroie** [comment]
The PCB v2.3 firmware is now available from the GitHub Actions tab.

**Comment by yabdali** [comment]
Is there any video on how to flash this device with an alternative firmware?

**Reply by cook1emr** [reply]
Hi, what are you struggling with? 

This device can be flashed through the usb port and using the web browser flashing.

**Reply by yabdali** [reply]
Do I need this step (IO0 pulled low (to GND) during power on to enter boot mode.) when using the USB flashing option? Where is the GPIO0 in the schematic, can't find it.

**Reply by htvekov** [reply]
GPIO0 is located in the J1 connector block marked red in the PCB pictures further up in this thread. Naming on PCB is IO0. And it's needed to pull GPIO0 to ground and power cycle the device in order for the esp to enter flash mode

**Reply by yabdali** [reply]
Sorry, just double checking if this is what you meant? Connecting each pinpoint and plugging the other end of the USB cable to the the PC to get into bootload mode then removing the wire connecting GND with GPIO0 and proceeding with the rest of the steps?
![image](https://github.com/HASwitchPlate/openHASP/assets/30695308/5408cf38-c574-4942-b414-fcb770e90d90)

**Reply by macfly92** [reply]
Hello, I left the wire between GND and GPIO0 throughout the flashing process on my side.

**Comment by npapagiannis** [comment]
Hi all,
Can someone shed some light on where to find the "GS-T3E ESP32 binary"?
I can't find anything (or don't know what's the correct one) [here](https://install.openhasp.com/)

**Reply by cook1emr** [reply]
Sure, head to https://nightly.openhasp.com you will find it halfway down the page

**Reply by npapagiannis** [reply]
Well, that worked! 
Thanks!

**Comment by macfly92** [comment]
Hello,
Everything works perfectly except for the backlight management by home assistant. In the Custom component I checked "Brightness level when plate is idle" with a low value and set the plate to "short idle" 5 and "long idle" 10.

In the mqtt config I didn't understand the "HA LWT Topic" parameter, I couldn't find anything in the documentation so I left it by default (homeassistant/status) but I don't see this topic in my mqtt_explorer, maybe that's where it comes from.

I can also see in the Telnet logs : 
#[13:04:33.957][77812/95288 18][52864/54932  4] MSGR: File not found: L:/idle_short.cmd
#[13:04:54.956][77812/95288 18][52864/54932  4] MSGR: File not found: L:/idle_long.cmd

There is also some strange states in MQTT topic : 
![image](https://github.com/HASwitchPlate/openHASP/assets/24654951/bde5a4be-632b-4e7d-a734-8dbffe0fad87)
I can see Idle = off, but idle = long in the status update ....

If anyone has any ideas, I'd love to hear from you :) 

Thanks!

**Comment by totkeks** [comment]
Just found this discussion while searching for ~wall panels~ switch plates (finally know how to call them in English) online. Thank you for sharing this product and it's EU compatibility as well as making me aware of openHASP, so I don't have to implement the UI myself. 😅

I have two questions from my initial readings:

1. Why can't the Zigbee chip be used? Is there no open/official documentation how to communicate with it?
2. Would it be possible to use openHASP on the bigger T8E (the linux one, not the Android one)? I guess the project lacks support for ARM chips and then the question is, should I just use an embedded Linux instead?

Thanks! :)

**Comment by 8OND007** [comment]
> Just found this discussion while searching for ~wall panels~ switch plates (finally know how to call them in English) online. Thank you for sharing this product and it's EU compatibility as well as making me aware of openHASP, so I don't have to implement the UI myself. 😅
> 
> I have two questions from my initial readings:
> 
> 1. Why can't the Zigbee chip be used? Is there no open/official documentation how to communicate with it?
> 2. Would it be possible to use openHASP on the bigger T8E (the linux one, not the Android one)? I guess the project lacks support for ARM chips and then the question is, should I just use an embedded Linux instead?
> 
> Thanks! :)

Would indeed be very interesting if we could use the zigbee chip also.
I just received my GS-T3E V2.3 today from Ali :-)
Just succesfully flashed it with OpenHASP. Opening the case went easy (Front from backplate went quick, if the plate is later attached to the wall back plate I suspect I will have to put my flat screw driver from below in those 2 slots to lift it of the wall back end plate,.
, then remove backside from frontplate). Lifting the PCB up to put in the micro USB to flash was a bit tricky, PCB was stuck bottom left side, so it was just not bended.
I just noticed the PCB cable from backplate to frontplate (5VDC power, GPIO's for switching the relays) has the same header cable type as the onces I bought for the WT32-SC01 plus debug header. Only the WT32-SC01 plus has 7 pins and this GS-T3E has 8 pins. But the cabled fitted. So I can use this cable to power the frontplate directly with 5VDC (only DC power in my wallbox).
It pretty bright the screen. The WT32-SC01 plus was broken, so I can't find the pages.jsonl file anymore on my home assistant since it was only stored on the display itself. I do still have the corresponding yaml file (side HA to communicate with it).
Maybe a dumb question, but does anyone already made a 3D print version of the backplate clip construction? So I could replace the included Power supply/relays block with a flat back plate that clamps onto the frontplate of the GS-T3E ?

**Reply by kstevn** [reply]
Hey, do you have a link to the item on aliexpress?

**Comment by Babass14** [comment]
Hello,
Anyone have the original firmware file for T3E with v2.3 PCB ?

**Reply by bensuffolk** [reply]
I have just received a new panel and not yet flashed it. Let me see if I can download the factory firmware for you later today.

**Reply by Babass14** [reply]
Thanks a lot.
Normally no problem to backup with esp tool (it's a 16mb memory).

**Reply by bensuffolk** [reply]
You may need to help me out with the command. I can see its a 16M

```
$ python3.9 esptool.py flash_id
esptool.py v4.4
Found 3 serial ports
Serial port /dev/cu.usbmodem14101
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (revision v0.2)
Features: WiFi, BLE
Crystal is 40MHz
MAC: xx:xx:xx:xx:xx:xx
Uploading stub...
Running stub...
Stub running...
Manufacturer: c8
Device: 4018
Detected flash size: 16MB
Hard resetting via RTS pin...
```

But it does not want to download it

```
$ python3.9 esptool.py read_flash 0x0 0x1000000 t3e.bin
esptool.py v4.4
Found 3 serial ports
Serial port /dev/cu.usbmodem14101
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (revision v0.2)
Features: WiFi, BLE
Crystal is 40MHz
MAC:xx:xx:xx:xx:xx:xx
Uploading stub...
Running stub...
Stub running...

A fatal error occurred: Invalid head of packet (0x6F): Possible serial noise or corruption.
```

--no-stub seems to help, but it always gives up with the same error message.

```
$ python3.9 esptool.py --no-stub read_flash  0x0 0x1000000 t3e.bin
esptool.py v4.4
Found 3 serial ports
Serial port /dev/cu.usbmodem14101
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (revision v0.2)
Features: WiFi, BLE
Crystal is 40MHz
MAC: xx:xx:xx:xx:xx:xx
Enabling default SPI flash mode...
2097152 (12 %)
A fatal error occurred: Failed to read flash block (result was 01090000: CRC or checksum was invalid)
```

It will download 8MB without any problems, but anything over that give the above errors.

```
$ python3.9 esptool.py read_flash 0x0 0x800000 t3e-16m.bin
esptool.py v4.4
Found 3 serial ports
Serial port /dev/cu.usbmodem14101
Connecting...
Detecting chip type... ESP32-S3
Chip is ESP32-S3 (revision v0.2)
Features: WiFi, BLE
Crystal is 40MHz
MAC: xx:xx:xx:xx:xx:xx
Uploading stub...
Running stub...
Stub running...
8388608 (100 %)
8388608 (100 %)
Read 8388608 bytes at 0x00000000 in 730.3 seconds (91.9 kbit/s)...
Hard resetting via RTS pin...
```

**Reply by nagyrobi** [reply]
Try with an external USB/UART adapter connected to the chip's RX/TX pin (like used to with the classic ESP32).

**Reply by bensuffolk** [reply]
I will try, but thats a bit more of a pain as the pinout does not match the default, so I will need to wait till I am home with some wires to do that, the adapter won't just plug in.

**Reply by nagyrobi** [reply]
I've had the same experience with an ESP32-S2, the built-in USB-UART was simply unable to properly dump the firmware.

**Reply by bensuffolk** [reply]
I have just tried the external USB UART and I get exactly the same results.

So if the first 8MB is useful let me know and you can have it, otherwise I'm sorry I can't be of help/

**Reply by Babass14** [reply]
I have tried to flash the openHASP firmware and just after to dump the firmware.
I have the same result (with external USB UART directly on RX/TX pin).
I could flash different firmware file but impossible to dump...

**Comment by Babass14** [comment]
Hello all,
Many thanks !
@bensuffolk , please send me .bin file for the 8mb.

**Reply by bensuffolk** [reply]
[t3e-0-8.bin.zip](https://github.com/HASwitchPlate/openHASP/files/14462593/t3e-0-8.bin.zip)

**Reply by Babass14** [reply]
I've tried to flash with your file, but no result on screen but it's normal (flash is ok with no error).
Thanks for your help.
I've asked the supplier to have the orginial file...

**Reply by deniselec** [reply]
I've read the original firmware with esptool, then flash OpenHasp and reflash the original and this work fine

**Comment by bensuffolk** [comment]
I have just added a USR-ES1 (Wiznet w5500) and its all working. Sadly there are not enough free pins on the existing breakout connector, and you need to solder some wires to the U2 pads.

I removed the Tuya chip, but you could just cut the power trace to it (after U2 as we need the 3.3v for the Ethernet module)

So next step for me is to design a PCB and make use of POE. I guess I will also need to 3D print a case for it as well. I think I saw some mounting plates somewhere for this that would be a good starting point, but for the life of me I can't find them now.

**Reply by bensuffolk** [reply]
![IMG_5709](https://github.com/HASwitchPlate/openHASP/assets/12720153/18c909ce-f19e-4cf8-bc1a-4aa1d180001e)

**Reply by totkeks** [reply]
What is that metal plate in the top right corner of the case? It wasn't there in the original photo of this post.

**Reply by bensuffolk** [reply]
I believe thats the back of the screen, it looks like in the original photo there is a bit of plastic covering it. I guess a slightly different case in the later units?

**Reply by fantasygoat** [reply]
@bensuffolk Did you ever make a PCB and case for this mod?

**Reply by bensuffolk** [reply]
I did, here is the PCB. https://github.com/HASwitchPlate/openHASP/discussions/933#discussioncomment-14385588 I have not been happy with the 3D case though, the screen does not clip onto it properly. I'm considering other options there

**Comment by aukselis-c1connections** [comment]
On Alibaba now I see they have 2024 version with a higher resolution OLED display. Has anyone  tried it already? Does it work with openHASP?
https://www.alibaba.com/product-detail/2024-More-stable-smart-home-switch_1601055512626.html?spm=a2700.shop_plser.41413.3.498c27aakmGsSu

**Reply by totkeks** [reply]
I don't see any difference there. Just a product listing with incorrect information.

The first higher resolution is wrong. Later it shows the product details for T6E. Further down they show the correct T3E information, with 480x480 IPS display.

No OLED, no higher resolution unfortunately.

**Reply by aukselis-c1connections** [reply]
@totkeks, thanks for the info. It is unfortunate indeed, got excited about the oled. Difficult to choose which one to buy as the one that is linked in the original post is not shipable anymore. And they have so many different listings of the same thing with slightly different prices.

**Reply by fvanroie** [reply]
Never trust any of the information on Alibaba/AliExpress listings to be correct...
even after asking a representative, they will just say whatever to get a sale.

Sadly the only way to find out is to order a sample. If you end up getting one from Alibaba and it works, please share the link so I can update it.

**Reply by aukselis-c1connections** [reply]
I bought this one: https://www.alibaba.com/product-detail/2022-new-arrival-smart-4-inch_1600560175304.html?spm=a2756.order-detail-ta-bn-b.0.0.33982fc2rw8xFa

I just received it and flashed openHASP. It seems to be working well. However, I am not able to dim the screen. Is the screen dimmable in this unit or am I doing something wrong?

**Reply by bensuffolk** [reply]
Dimming does not work for me, so I assumed that its not supported by the hardware. I think from my trials once you get below around 200 it just cuts off.

What I did was have a full sized transparent box that I enable when I get the IDLE status sent to my MQTT server. Then it appears to dim the screen from the users perspective. When the screen is touched the IDLE off status comes in and I hide the box.

**Comment by dantyr2** [comment]
can I have your gs-t3e.json file? I just want to se how it looks as an excemple :)

**Reply by bensuffolk** [reply]
If that was aimed at me in reference to the transparent box in the thread above, the relevant lines would be:

`{"page":0,"id":0,"click":0}
{"id":255,"obj":"obj","x":0,"y":0,"w":480,"h":480,"bg_opa":192,"bg_color":"black","border_side":0,"hidden":1,"click":0}
`

Then show object p0b255 when you want the screen dimmed (when you get an IDLE command), it will click through so anything underneath will still work. When you get the IDLE off command make sure you hide object p0b255.

**Comment by polobakehouse** [comment]
hi! how can install home assistant in this device?

**Reply by cook1emr** [reply]
You can't install home assistant on the device. You install custom firmware that talks to home assistant via mqtt. This is all detailed in the openhasp documentation

**Comment by 2low4slow** [comment]
hello everybody,
the relay of the device is not autodetect by homeassistant, may anyone can help me?

**Reply by cook1emr** [reply]
In the device do you have any entities to enable?

![Screenshot_20241216-071334_Home Assistant.jpg](https://github.com/user-attachments/assets/47a558f8-0471-4575-bbdc-b8e8c11dc1dc)

You should see light 44, 45 and 46

However you need to enable the gpio in openhasp first

![20241216_071758.jpg](https://github.com/user-attachments/assets/5090fee2-1355-4777-97ff-2529e5dc5be8)

**Reply by 2low4slow** [reply]
i already enable the GPIO like written in the documentation
![Bildschirmfoto 2024-12-16 um 09 05 48](https://github.com/user-attachments/assets/c0cacf8d-f687-4b6d-b240-4fd4b6595a27)
but all i see in homeassitant is this
![Bildschirmfoto 2024-12-16 um 09 05 11](https://github.com/user-attachments/assets/16015991-5853-4440-b0b8-7f2824cf7e85)

**Reply by fvanroie** [reply]
Remove the device in home-assistant and rediscover the device. That should add the configured relays in home-assistant too.

**Reply by 2low4slow** [reply]
THX - that solved the problem

**Comment by unmo3d** [comment]
does she work with ESPHOME?

**Comment by Meri61** [comment]
Hi everyone,
Please Help! I have purchased today the Zigbee one T3e (I think the newest Version). I have tried to connect it with Tuya app integration in HA. It was successful but has no entities! I can not switch my light with it. I may configure it as here discussed? If yes can someone sen me a link to a step by step guide? I am really newbie to this subject. Thank u so much!

**Comment by kash1982** [comment]
what firmware is everyone using to flash this?

**Reply by cook1emr** [reply]
https://nightly.openhasp.com/ or https://github.com/HASwitchPlate/openHASP/releases/download/0.7.0-rc13/globalsecurity.zip

**Reply by kash1982** [reply]
Thank you

**Comment by kash1982** [comment]
has anyone had any luck with board revision 2.4 with no no micro usb?

ive read the flash but cant write to the device using the global security bin

i get the following

Uploading stub...
Running stub...
Stub running...
Changing baud rate to 115200
Changed.
NO XMC flash  detected!
SPI_BOOT_CRYPT_CNT 0
SECURE_BOOT_EN False

chip type is:

Chip is ESP32-S3 (QFN56) (revision v0.2)
Features: WiFi, BLE, Embedded PSRAM 8MB (AP_3v3)
Crystal is 40MHz
MAC: d83bda700334
Manufacturer: c8
Device: 4018
Status value: 0x200200
Detected flash size: 16MB

i have dumped the flash also

pictures:


![20250404_142040](https://github.com/user-attachments/assets/a3331220-25dc-43d6-88eb-a14612e69443)
![20250404_142054](https://github.com/user-attachments/assets/da51c2d1-7ad6-4140-87f0-66cc9f732876)
![20250404_142100](https://github.com/user-attachments/assets/6dde18d2-7e07-4f6c-8c32-d8ec3dc5535c)

**Reply by cook1emr** [reply]
I have just had a few of these versions turn up. Am a little disappointed the are the v2.4 rather than v2.3.

I think they are the version discussed here https://github.com/HASwitchPlate/openHASP/discussions/458#discussioncomment-9019672

**Reply by kash1982** [reply]
Let me know how you get on. I've not been able to flash it but I'm going to try on a different PC tomorrow.

**Comment by kash1982** [comment]
flashed today perfectly. for the 2.4 revision tx and rx are as standard and crossed over.

TX - RX
RX -TX

**Reply by cook1emr** [reply]
How are you finding the device? Is the wifi good?

**Reply by kash1982** [reply]
Wifi is perfect. I do have it mounted in the same room as one of our ap's however

**Reply by trav13k** [reply]
Can also confirm the above for v2.4. Flashed perfectly once you referse the TX and RX from the original description...

**Reply by Jero1985** [reply]
Hello, has anyone managed to get version 2.4 to work completely? I'm currently failing at the touch. I only program with AI and ESPHOME because I have no idea ;-)

**Comment by sleepypig2003** [comment]
This is my experience with the newer T3E from Golden Security.
I bought the T3E 4-gang relay version (China) with the v2.4 board  ( ZIGBEE_4CP_WNB_V2.4 - 2024.09.09 )

**Hardware**

There are 4 relays:  L, N, L1, L2, L3, L4

![4_gang_relay_board](https://github.com/user-attachments/assets/a57716bf-3e6d-46df-a780-96330f89b854)
![4_gang_relay_board_back](https://github.com/user-attachments/assets/66c4c4ce-8acf-4eba-bdd0-f0773d41e34f)

**Flashing**

The v2.4 board no longer comes with a USB socket.  Fortunately the programming connections are still there.

Connecting from the T3E v2.4 <--> FTDI serial programmer is as follows (this is normal, unlike v2.3 which mislabelled RX and TX)
GND <--> GND
5V <--> 5V
RX <--> TX
TX <--> RX
IO0 <--> GND during power on to enter boot mode

![v2 4_programming_pins](https://github.com/user-attachments/assets/6eed2e6e-061f-476e-a970-d5cf48c28b64)

If you don't want to solder, another option is to use pogo pins to read/write the ESP32 flash.

I used the following 6 pin pogo pin clamp which saves me from soldering and de-soldering:

Test Stand PCB Clip 2.54mm Spacing Clamp Burning Fixture Probe Pogo Pin Download Program Burn 6P
https://www.aliexpress.com/item/1005006108783889.html

![v2 4_programming_clamp](https://github.com/user-attachments/assets/b5a1b017-9d15-4001-9980-0b39ac34056f)


**GPIO configuration**

GPIO1 = L1 relay
GPIO2 = L2 relay
GPIO46 = L3 relay
GPIO45 = L4 relay

![image](https://github.com/user-attachments/assets/cd813bd7-4a18-4147-a63b-fd626dc888e7)

**Reply by cook1emr** [reply]
My v2.4 are still in the box. Havent had a chance to play with them yet. Have you found any additional spare gpio's available?

**Comment by GrzegorzRussek** [comment]
Did anyone tinker with this version (ordered from same seller and always got V2.4, but this is new one)

![IMG_20250615_130317917_HDR](https://github.com/user-attachments/assets/236f9001-86e0-4b73-9dca-87fd9f5769e6)

**Reply by cook1emr** [reply]
not seen that one, but this looks like bad news. I cant see the microcontroller labels in your photos, but doesn't look like they are esp32's

**Reply by fvanroie** [reply]
As this is not an ESP32 device, openHASP will not run on it.
Is there a way to tell the device is a T3E-Pro based on the box or labeling?
We can add a warning to the docs. Did you get the device on Alibaba or AliExpress?

**Comment by lexmark3200** [comment]
@kash1982 , @sleepypig2003 
What programmer did you use for version 2.4? I have a CP2102 and I'm not willing to talk about it. Here's a link to the discussion: https://github.com/HASwitchPlate/openHASP/discussions/1026#discussion-10307897
